From 173a9a51f4d79a9d76987a9e425a64038c887b3e Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Mon, 4 Apr 2011 11:16:16 +0200
Subject: [PATCH] patch: xsane_update_param-crash

Squashed commit of the following:

commit 4b2f69d1b51db7160a33764a4f68ab3c4fc43678
Author: Nils Philippsen <nils@redhat.com>
Date:   Mon Apr 4 11:13:20 2011 +0200

    don't dereference xsane.preview if it is NULL
---
 src/xsane-front-gtk.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/xsane-front-gtk.c b/src/xsane-front-gtk.c
index 08761a0..66a3177 100644
--- a/src/xsane-front-gtk.c
+++ b/src/xsane-front-gtk.c
@@ -1713,7 +1713,7 @@ void xsane_update_param(void *arg)
   gtk_label_set(GTK_LABEL(xsane.info_label), buf);
 
 
-  if (xsane.preview->surface_unit == SANE_UNIT_MM)
+  if (xsane.preview && xsane.preview->surface_unit == SANE_UNIT_MM)
   {
    double dx, dy;
 
-- 
1.7.4.2

